src/main/java/com/puppycrawl/tools/checkstyle/JavaAstVisitor.java

```diff
@@ -1336,7 +1335,7 @@ public DetailAstImpl visitSuperExp(JavaLanguageParser.SuperExpContext ctx) {
             superSuffixParent = bop;
         }
         else {
-            DetailAstImpl firstChild = superSuffixParent.getFirstChild();
+            DetailAstImpl firstChild = superSuffixParent;
```

Comment:
Similar reasoning to https://github.com/checkstyle/checkstyle/pull/12588#issue-1520971449 . The loop keeps going down through the children until it finds a `null` first child. This just adds 1 more loop processing time since it doesn't automatically go into the first child, however, this prevents a possible (but probably unlikely) NPE that `superSuffixParent`'s first child is `null` already.